automated-actions/AWS_ELASTICLOADBALANCING_ENI_LIMIT_REACHED/LambdaFunction.js

```diff
@@ -0,0 +1,75 @@
+// Sample Lambda Function to remove unattached ENIs in the region of the event when AWS Health AWS_ELASTICLOADBALANCING_ENI_LIMIT_REACHED events are generated. 
+// This is useful for situations where you might have leftover ENIs that are not used and are preventing load balancer scaling
+'use strict';
+var AWS = require('aws-sdk');
+const dryRun = ((process.env.DRY_RUN || 'true') == 'true');
+const maxEniToProcess = process.env.MAX_ENI || 100;
+
+//main function which gets AWS Health data from Cloudwatch event
+exports.handler = (event, context, callback) => {
+    //extract details from Cloudwatch event
+    var eventName = event.detail.eventTypeCode;
+    var region = event.region;
+    const awsHealthSuccessMessage = `Successfully got details from AWS Health event ${eventName} and executed automated action in ${region}. Further details in CloudWatch Logs.`;
+
+    // we only need to run this automation once per invocation since the issue 
+    // of ENI exhaustion is regional and not dependent on the load balancers in the alert
+    // Event will only trigger for one region so we don't have to loop that
+    AWS.config.update({region: region});
+    AWS.config.update({maxRetries: 3});
+    var ec2 = new AWS.EC2();
+    
+    console.log ('Getting the list of available ENI in region %s', region);
+    var params = {
+        Filters: [{Name: 'status',Values: ['available']}]
+    };
+    
+    ec2.describeNetworkInterfaces(params, function(err, data) {
+        if (err) 
+        {
+            console.log( region, err, err.stack); // an error occurred
+            callback('Error describing ENIs; check CloudWatch Logs for details');
+        }
+        else 
+        {
+			var numberToProcess = data.NetworkInterfaces.length;
+			if ((maxEniToProcess > 0) && (data.NetworkInterfaces.length > maxEniToProcess)) numberToProcess = maxEniToProcess;
+            console.log('Found %s available ENI; processing %s',data.NetworkInterfaces.length,numberToProcess);
+            // for each interface, remove it
+            for ( var i=0; i < numberToProcess; i+=1)
+            {
+                deleteNetworkInterface(data.NetworkInterfaces[i].NetworkInterfaceId,dryRun); 
+            }
+            
+            callback(null, awsHealthSuccessMessage); //return success
+        }
+    });
+};
+
+//This function removes an ENI
+function deleteNetworkInterface (networkInterfaceId, dryrun) {
+    var ec2 = new AWS.EC2();
```

Comment:
Hmm - I'll take a look at that. At line #6 I don't yet have the region of the event though - so going to have to see if there is a way to set the region AFTER creating the object.